crates/viewer/re_viewer_context/src/selection_state.rs

```diff
@@ -391,47 +91,39 @@ impl ApplicationSelectionState {
         re_tracing::profile_scope!("SelectionState::on_frame_start");
 
         // Purge selection of invalid items.
-        let selection_this_frame = self.selection_this_frame.get_mut();
-        selection_this_frame.retain(|item, _| item_retain_condition(item));
-        if selection_this_frame.is_empty()
+        self.selection.retain(|item, _| item_retain_condition(item));
```

Comment:
very subtle bug here: previously, if we purged a single element from a multi-selection, we'd fire a selection change event. Now, we only fire it if we purged all of them.
Have to check count before/after `retain` to set `selection_changed`

Reply:
Ahh yeah good catch!